### PR TITLE
Add spotter application flow to the app

### PR DESCRIPTION
**Pull request by tleraj**
The purpose of this PR is to establish the new application flow in the frontend. The designs implemented look like this:

Screenshot from 2020-10-07 14-59-44
Screenshot from 2020-10-07 14-59-27

### CODE REVIEW

File: packages/website/src/routes/Apply/__snapshots__/index.test.tsx.snap [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`renders as expected 1`] = `
+exports[`Apply Page should render with given state from Redux store 1`] = `
```
Comment (ericboucher) [inline packages/website/src/routes/Apply/__snapshots__/index.test.tsx.snap:3]: Site registration page

File: packages/website/src/routes/Apply/index.tsx [hunk 2]
```diff
@@ -56,10 +60,32 @@ const agreements = [
...
+          .set("name", user.fullName)
+          .set("org", user.organization)
+          .set("email", user.email)
+      );
+    } else {
+      setFormModel(formModel.set("name", "").set("org", "").set("email", ""));
+    }
+  }, [user, formModel]);
 
   const locationFormElements = [
```
Comment (ericboucher) [inline packages/website/src/routes/Apply/index.tsx:90]: Maybe move the reef name to here
  { id: "siteName", label: "Site Name" },


And disable all the contactForm elements instead?

File: packages/api/src/reef-applications/dto/create-reef-application.dto.ts [hunk 3]
```diff
@@ -30,6 +30,9 @@ export class CreateReefApplicationDto {
 }
 
 export class CreateReefWithApplicationDto {
+  @IsNotEmpty()
```
Comment (ericboucher) [inline packages/api/src/reef-applications/dto/create-reef-application.dto.ts:33]: Is this restriction also enforced on the frontend?

Comment (tleraj) [reply packages/api/src/reef-applications/dto/create-reef-application.dto.ts:33]: Yes it is a required field.

File: packages/website/src/routes/Apply/index.tsx (hunk 2 above)
Comment (tleraj) [reply packages/website/src/routes/Apply/index.tsx:82]: Then the fields will be in different order than the ones on the doc https://docs.google.com/document/d/1v8TZW2H74R2isCH-Uf57JkDkxoRYt2NjuXvLPpo7RTI/edit.

Comment (ericboucher) [reply packages/website/src/routes/Apply/index.tsx:82]: I think that's fine

**Comment by ericboucher** [comment]
The new reef appears in my profile list as expected but it looks like we need to add a few guards to avoid errors if we try loading the reef too early:

Screen Shot 2020-10-07 at 5 45 05 PM

File: packages/website/src/routes/Apply/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    value: 26,
+    timestamp: "2020-07-01T14:25:18.008Z",
+  },
+  degreeHeatingDays: {
+    value: 32,
+    timestamp: "2020-07-01T14:25:18.008Z",
+  },
+};
+
+const dailyData = {
```
Comment (ericboucher) [inline packages/website/src/routes/Apply/index.test.tsx:28]: do you need this data in here?

Comment (tleraj) [reply packages/website/src/routes/Apply/index.test.tsx:28]: Just using it to mock a reef extensively.

File: packages/website/src/helpers/findAdministeredReef.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { User } from "../store/User/types";
+import { Reef } from "../store/Reefs/types";
+
+export const findAdministeredReef = (
+  user: User | null,
+  reefId: number
+): Reef | undefined => {
+  const reef = user?.administeredReefs?.find((item) => item.id === reefId);
```
Comment (ericboucher) [inline packages/website/src/helpers/findAdministeredReef.ts:8]: no need for a declaration here, just return

File: packages/website/src/routes/Apply/Form/index.tsx [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+  agreed,
+  handleFormSubmit,
+  classes,
+}: FormProps) => {
+  const [installationSchedule, setInstallationSchedule] = useState<Date | null>(
+    null
+  );
+
+  useEffect(() => {
+    if (application && application.installationSchedule) {
```
Comment (ericboucher) [inline packages/website/src/routes/Apply/Form/index.tsx:32]: let's use chaining

**Review by ericboucher — APPROVED** [review]
LGTM with minor changes

**Pull request merged**